crates/yellowstone-fumarole-source/src/lib.rs

```diff
@@ -50,26 +60,63 @@ impl From<FumaroleConfig> for yellowstone_fumarole_client::config::FumaroleConfi
 impl YellowstoneFumaroleSource {
     /// Create a new `YellowstoneFumaroleSource` with default values.
     #[must_use]
-    pub fn new(subscriber_name: &str) -> Self {
-        Self {
-            config: FumaroleConfig {
-                subscriber_name: subscriber_name.to_string(),
-                base: None,
-            },
-            filters: None,
-        }
+    pub fn new() -> Self { Default::default() }
+
+    #[cfg(feature = "prometheus")]
+    fn register_metrics(&self, config: &FumaroleConfig) -> JoinHandle<()> {
+        use std::time::Duration;
+
+        let prometheus_registry = prometheus::Registry::new();
+        yellowstone_fumarole_client::metrics::register_metrics(&prometheus_registry);
+
+        // Use the conditional config fields
+        let export_interval = Duration::from_secs(config.metrics_interval);
+        let job_name = config.metrics_job_name.clone();
+        let metrics_endpoint = config.metrics_endpoint.clone();
+
+        // Spawn metrics pusher task
+        let registry_clone = prometheus_registry.clone();
+        tokio::spawn(async move {
+            let mut interval = tokio::time::interval(export_interval);
+            loop {
+                interval.tick().await;
+
+                let metrics = registry_clone.gather();
+                let job_name = job_name.clone();
+                let metrics_endpoint = metrics_endpoint.clone();
+                let _ = tokio::task::spawn_blocking(move || {
+                    if let Err(e) = prometheus::push_metrics(
+                        &job_name,
+                        prometheus::labels! {},
+                        &metrics_endpoint,
+                        metrics,
+                        None,
+                    ) {
+                        tracing::error!("Failed to push Fumarole metrics: {e:?}");
+                    }
+                })
+                .await;
+            }
+        })
```

Comment:
This is designed where if the binary using vixen when it has prometheus feature active on fumarole will export its metrics to the endpoint of the app. We don't need to expose our own. 

This is how I'd like our metrics to work too. We just support prometheus as a feature with no push gateway. The user has to setup their own server.